### ISSUE TITLE
Crash when connecting to latest licode server

### ISSUE THREAD

**Issue by vrevidat** [zevarito/ErizoClientIOS#22]
Hi.  

I installed the latest Licode on one of my servers. The iOS client crashes when connecting to it.
It seems to be caused by this unknown type received in the websocket event:

??DEBUG??: Websocket didReceiveEvent "{"name":"signaling_message_erizo","args":[{"mess":{"type":"started"},"streamId":570008531212806.8}]}"
2016-04-05 18:34:59.696 ECIExampleLicode webrtc[2476:1124912] 
Source: /ErizoClient/rtc/ECSignalingMessage.m:84
??WARN??: Unexpected type: started 
2016-04-05 18:34:59.696 ECIExampleLicode webrtc[2476:1124912] *** Assertion failure in -[ECSignalingChannel signalingDelegateForStreamId:], /ErizoClient/rtc/ECSignalingChannel.m:330
2016-04-05 18:34:59.702 ECIExampleLicode webrtc[2476:1124912] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'streamId needs to be a string'
*** First throw call stack:
(0x180e7ee38 0x1804e3f80 0x180e7ed08 0x181804124 0x1004b3674 0x1004b1588 0x1004a145c 0x10049d3f4 0x100495b98 0x100a75a7c 0x100a75a3c 0x100a7b4e4 0x180e34dd8 0x180e32c40 0x180d5cd10 0x182644088 0x186031f70 0x1000bfa08 0x1808fa8b8)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 

Full trace attached....
[trace_ios_crash.txt](https://github.com/zevarito/ErizoClientIOS/files/205496/trace_ios_crash.txt)

Thanks a lot for the help...

+vrevidat

**Comment by zevarito** [comment]
Hi @vrevidat,

Can you place a debugger before this line: https://github.com/zevarito/ErizoClientIOS/blob/master/ErizoClient/rtc/ECSignalingChannel.m#L330
and then see what is `streamId` there? Also all the way back where it is assigned, it should be from server side response.

You can safely ignore the warning message about `Unexpected type: started `, since even JS client seems to implement it: https://github.com/ging/licode/blob/1603b6f19de1278f3ec8a0a8ee992045f2a60568/erizo_controller/erizoController/roomController.js#L104

Let me know how it works.

**Comment by vrevidat** [comment]
Thank you for the quick answer :)

`streamId` is nil.

Data i am getting from the server is:

`5:::{"name":"signaling_message_erizo","args":[{"mess":{"type":"started"},"streamId":43151393765583630}]}`

Thanks a lot

**Comment by zevarito** [comment]
@vrevidat I've tried with the Licode version I have installed and not publishing nor consuming causes that messages to be sent. I am some commits behind Licode master branch, I guess the protocol changed, of course it isn't documented, at least last time I've checked.

I'll eventually fix when update Licode but I cannot tell you when it will happen, meanwhile if you want to take this own your own, here is the flow I'd suggest.

* Open Chrome init/consume a stream while debugging WebSockets to realize which is the protocol both parts are talking and where exactly `type: started` takes place and for what it is used.

* Then update this method to handle that message type: https://github.com/zevarito/ErizoClientIOS/blob/master/ErizoClient/rtc/ECSignalingMessage.m#L52

* If everything looks good, PR are welcome.

This is only a quick suggestion about what I would try first, but honestly don't have looked deeply on the problem.

Good luck!

**Comment by ashar02** [comment]
I have already fixed this problem in my code. It does not handle the message fully but at-least it does not crashes, if these new un-handled messages arrives. Send you PR to you.

**Comment by zevarito** [comment]
@ashar02 that's cool, did you figure out for what is exactly used?

**Comment by vrevidat** [comment]
I did some research this week end and was not able to understand when this started parameter was introduced and why.  Connecting to my old licode server, for each stream subscribed i am getting "answer" and "ready".  With the new server, i am getting "started", "answer" and "ready".  Still no clue why, but i will continue investigations.  In the meantime I added a check for nil in the calling function and it's not crashing and working fine.
